### DISCUSSION TITLE
API request for STT works fine with specifying a URL, but not when base64 encoding the file

### DISCUSSION THREAD

**Discussion by ewheelerinc** [deepgram/community#428] [General help] [answered]
We are testing out the Deepgram API for STT and are having an issue getting base64-encoded audio requests to work. However,  it works fine when we specify a URL.

These are the two requests:

-= WORKS =-
`curl --request POST \
    --url 'https://api.deepgram.com/v1/listen?filler_words=false&summarize=v2' \
    --header 'Authorization: Token XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX' \
    --header 'accept: application/json' \
    --header 'content-type: application/json' \
    --data '
        {
            "url": "https://www.linuxglobal.com/out/t.wav"
        }
    '
`

-= DOES NOT WORK =-
`
BASE64=$(base64 -w0 /tmp/t.wav)

curl --request POST \
    --url 'https://api.deepgram.com/v1/listen?filler_words=false&summarize=v2' \
    --header 'Authorization: Token 449eff8d18907a56b66b8f1ff2df2c8d08690f92' \
    --header 'accept: application/json' \
    --data "data:audio/x-wav;name=t.wav;base64,$BASE64"
`

Result:

`
{
  "err_code" : "Bad Request",
  "err_msg" : "Bad Request: failed to process audio: corrupt or unsupported data",
  "request_id" : "63b79b92-e8f7-43b2-8c49-b50def803954"
}
`

What are we doing wrong? Note that the demo (https://developers.deepgram.com/reference/pre-recorded) also has "--header 'content-type: application/json'" in the cURL request, but that just results in a different error:

`
{
  "err_code" : "Bad Request",
  "err_msg" : "Content-type was application/json, but we could not process the JSON payload.",
  "request_id" : "0e5d9bf1-af75-40da-83a2-cbdcc8526fdf"
}
`

It is also worth noting that the cURL requests are pulled from the demo itself, and I even tried running it from the demo and it just is not working unless I specify a URL, but we need file-based transcript generation, not URL-based.

**Comment by DamienDeepgram** [comment, marked as answer]
You can find how to transcribe a local file here: https://developers.deepgram.com/docs/transcribing-pre-recorded-audio#transcribe-a-local-file

> curl \
  --request POST \
  --header 'Authorization: Token YOUR_DEEPGRAM_API_KEY' \
  --header 'Content-Type: audio/wav' \
  --data-binary @youraudio.wav \
  --url 'https://api.deepgram.com/v1/listen'

Also make sure to rotate the API Key you posted above

**Reply by ewheelerinc** [reply]
That worked, thanks! It is interesting that the demo with pre-recorded does not provide the proper POST request. You might update the demo code so that it works when uploading a file.

Question: Is there a content-type for raw MULAW?
